runtime/src/main/scala/akka/grpc/scaladsl/headers/headers.scala

```diff
@@ -81,19 +86,86 @@ object `Status` extends ModeledCustomHeaderCompanion[`Status`] {
 }
 
 object `Status-Message` extends ModeledCustomHeaderCompanion[`Status-Message`] {
   override val name = "grpc-message"
   override val lowercaseName: String = super.lowercaseName
 
-  override def parse(value: String): Try[`Status-Message`] = Try(new `Status-Message`(value))
+  override def parse(value: String): Try[`Status-Message`] = Try(new `Status-Message`(parseAsciiString(value)))
 
   def findIn(headers: immutable.Seq[HttpHeader]): Option[String] =
     headers.collectFirst { case h if h.is(name) => h.value() }
+
+  private val HexArr: Array[Byte] =
+    Array('0', '1', '2', '3', '4', '5', '6', '7', '8', '9', 'A', 'B', 'C', 'D', 'E', 'F')
+
+  // Copied with slight adaptations from https://github.com/grpc/grpc-java/blob/79e75bace40cea7e4be72e7dcd1f41c3ad6ee857/api/src/main/java/io/grpc/Status.java#L626
+  private def parseAsciiString(value: String): String = {
+    val transferEncoding = StandardCharsets.US_ASCII
+    // This can be made faster if necessary.
+    val source = value.getBytes(transferEncoding)
+    val buf = ByteBuffer.allocate(source.length)
+    var i = 0
+    while (i < source.length) {
+      if (source(i) == '%' && i + 2 < source.length) {
+        try {
+          buf.put(Integer.parseInt(new String(source, i + 1, 2, transferEncoding), 16).toByte)
+          i += 3
+        } catch {
+          case _: NumberFormatException =>
+            buf.put(source(i))
+            i += 1
+        }
+      } else {
+        buf.put(source(i))
+        i += 1
+      }
+    }
+    new String(buf.array, 0, buf.position(), StandardCharsets.UTF_8)
+  }
+
+  // Copied with slight adaptations from https://github.com/grpc/grpc-java/blob/79e75bace40cea7e4be72e7dcd1f41c3ad6ee857/api/src/main/java/io/grpc/Status.java#L582
+  private def toAsciiString(value: String): String = {
+    val valueBytes = value.getBytes(StandardCharsets.UTF_8)
+    val firstIndexToEscape = valueBytes.indexWhere(isEscapingChar(_))
+    if (firstIndexToEscape == -1)
+      new String(valueBytes)
+    else
+      toAsciiStringSlow(valueBytes, firstIndexToEscape)
+  }
+
+  private def isEscapingChar(b: Byte): Boolean = b < ' ' || b >= '~' || b == '%'
+
+  private def toAsciiStringSlow(valueBytes: Array[Byte], riArg: Int): String = {
+    var ri = riArg
+    val escapedBytes = new Array[Byte](riArg + ((valueBytes.length - riArg) * 3))
+    // copy over the good bytes
+    if (riArg != 0) System.arraycopy(valueBytes, 0, escapedBytes, 0, riArg)
+    var wi = ri
+
+    while (ri < valueBytes.length) {
+      val b = valueBytes(ri)
+      // Manually implement URL encoding, per the gRPC spec.
+      if (isEscapingChar(b)) {
+        escapedBytes.update(wi, '%')
+        escapedBytes.update(wi + 1, HexArr((b >> 4) & 0xf))
+        escapedBytes.update(wi + 2, HexArr(b & 0xf))
+        wi += 3
+      } else {
+        escapedBytes.update(wi, b)
+        wi += 1
+      }
+      ri += 1
+    }
+    new String(jutil.Arrays.copyOf(escapedBytes, wi))
```

Comment:
Should be equivalent to the following, and being explicit for the charset is better instead of relying of the system's default:
`new String(escapedBytes, 0, wi, StandardCharsets.US_ASCII)`